database.cc

```diff
@@ -1426,7 +1426,14 @@ database::query(schema_ptr s, const query::read_command& cmd, query::result_opti
             std::move(cache_ctx)).then_wrapped([this, s = _stats, &semaphore, hit_rate = cf.get_global_cache_hit_rate(), op = cf.read_in_progress()] (auto f) {
         if (f.failed()) {
             ++semaphore.get_stats().total_failed_reads;
-            return make_exception_future<std::tuple<lw_shared_ptr<query::result>, cache_temperature, query::status>>(f.get_exception());
+            try {
+                f.get();
+            } catch (exceptions::overloaded_exception& e) {
+                return make_ready_future<std::tuple<lw_shared_ptr<query::result>, cache_temperature, query::status>>(std::tuple(make_lw_shared<query::result>(), hit_rate, query::status::OVERLOADED));
```

Comment:
Note this is not backwards compatible. If an old node will get this it will consider it as empty value not as failed read.

Reply:
Oops, right. I guess that piggy-backing on an existing tuple won't work for upgrades then, and a new verb would be a way to go...

Reply:
Or, I can just introuduce a cluster feature.

Reply:
Different verb will require cluster feature anyway.